### PR TITLE
LTF-134 - Validate passwords

### PR DESCRIPTION
**Pull request by JHjava**
## What?

- Validate that the password is not empty.
- Validate that the retyped password is not empty.
- Validate that the password is at least 8 characters long.
- Validate that the password contains a number.
- Validate that the 2 password fields match
- Depending on where the validation fails, add the correct error message in the format outlined in the GOV.UK design system. The error messages have been documented here - https://docs.google.com/document/d/190S1eOHxrDKJdZDpYg2qJx5ObRHycU4oCVECOm5JoTo/edit
- Switch off HTML5 validation on the form as advised in the design system docs https://design-system.service.gov.uk/patterns/validation/

## Why?

- So we can be confident that the user has selected a secure password

### CODE REVIEW

File: src/main/java/uk/gov/di/services/ValidationService.java [hunk 1]
```diff
@@ -20,4 +22,24 @@
         }
         return emailErrors;
     }
+
+    public Set<PasswordValidation> validatePassword(String password, String retypedPassword) {
```
Comment (mrwilson) [inline src/main/java/uk/gov/di/services/ValidationService.java:26]: In principle, all looks good. We could code a bit more defensively here - if either of the inputs are `null` then this method will NPE.

In practice, this _shouldn't_ happen if we're passing them through from the Dropwizard inputs (I think it turns them into empty strings) but an extra clause to cover our bases, perhaps?

Comment (JHjava) [reply src/main/java/uk/gov/di/services/ValidationService.java:26]: Yep you're right, it does convert them to an empty String. But not a bad shout, I'll add something in.

Comment (JHjava) [reply src/main/java/uk/gov/di/services/ValidationService.java:26]: I've updated and added a test in for good measure

**Comment by sanisc** [comment]
Generally seems fine overall.

**Review by sanisc — APPROVED** [review]

**Pull request merged**